languages/tree-sitter-stack-graphs-typescript/build.rs

```diff
@@ -0,0 +1,98 @@
+// -*- coding: utf-8 -*-
+// ------------------------------------------------------------------------------------------------
+// Copyright © 2024, stack-graphs authors.
+// Licensed under either of Apache License, Version 2.0, or MIT license, at your option.
+// Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
+// ------------------------------------------------------------------------------------------------
+
+use std::collections::HashSet;
+use std::path::Path;
+
+use anyhow::{Result, anyhow, bail};
+use regex::Regex;
+
+/// A stack of dialects as selected by the directives
+#[derive(Debug, Default)]
+struct DialectStack(Vec<HashSet<String>>);
```

Comment:
Ack. My only thought was to maybe use this mechanism to also unify typescript and javascript one day. We have mixed language projects and plan to just use typescript TSG for both but it has some downsides.

Reply:
Understood, and I could see this become useful once we get there. Even in that scenario it might be enough to allow multiple dialects after the `#dialect` but disallow nesting, meaning we could still do without maintaining a stack. But I think this is easier to assess if there's concrete proposal re. a unified spec.